lib/FIR/afforestation.cc

```diff
@@ -25,9 +25,13 @@
 
 namespace Fortran::FIR {
 namespace {
-Expression *ExprRef(const parser::Expr &a) { return &a.typedExpr.get()->v; }
+Expression *ExprRef(const parser::Expr &a) {
```

Comment:
I know this PR maybe shouldn't mess with APIs, but I think that this function should return a reference rather than a pointer.

Reply:
Good idea. I've made that change and it does simplify things.
I also realized there should be an overloading that takes parser::Variable so I added that too.